lib/formats/kms.js

```diff
@@ -0,0 +1,288 @@
+// Copyright 2017 Joyent, Inc.
+
+module.exports = {
+	listKeys: listKeys,
+	read: read,
+	verify: verify,
+	sign: sign,
+	write: write,
+	toBuffer: toBuffer,
+};
+
+var assert = require('assert-plus');
+var SSHBuffer = require('../ssh-buffer');
+var crypto = require('crypto');
+var Buffer = require('safer-buffer').Buffer;
+var algs = require('../algs');
+var Key = require('../key');
+var ecdsa = require('./pkcs1')
+var PrivateKey = require('../private-key');
+var Identity = require('../identity');
+var rfc4253 = require('./rfc4253');
+var Signature = require('../signature');
+var utils = require('../utils');
+var Certificate = require('../certificate');
+var asn1 = require('asn1')
+const { KMSClient, ListKeysCommand, GetPublicKeyCommand, SignCommand } = require("@aws-sdk/client-kms");
+const pkcs1 = require('./pkcs1');
+
+function verify(cert, key) {
+	/*
+	 * We always give an issuerKey, so if our verify() is being called then
+	 * there was no signature. Return false.
+	 */
+	return (false);
+}
+
+let TYPES = {
+	'user': 1,
+	'host': 2
+};
+Object.keys(TYPES).forEach(function (k) { TYPES[TYPES[k]] = k; });
+
+let ECDSA_ALGO = /^ecdsa-sha2-([^@-]+)-cert-v01@openssh.com$/;
+
+async function listKeys() {
+	const client = new KMSClient();
+	const command = new ListKeysCommand();
+
+	return await client.send(command)
+		.then(data => {
+			console.info(data);           // successful response
+		})
+		.catch(err => {
+			console.trace(err); // an error occurred
+		});
+}
+
+/**
+ * Get the public key from a KMS key ARN
+ * @param keyArn
+ * @returns {Promise<Key>}
+ */
+async function read(keyArn, _) {
+	const client = new KMSClient();
+	const command = new GetPublicKeyCommand({ KeyId: keyArn });
+
+	return await client
+		.send(command)
+		.then(data => {
+			const publicKey = data.PublicKey;
```

Comment:
A few suggestions here but don't consider them blocking

```suggestion
		.then(({ PublicKey: publicKey } )=> { => {
```

Reply:
I actually like this.